src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -790,6 +793,7 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id,
       [this, actor_id, actor_table_data](Status status) {
         RAY_CHECK_OK(gcs_publisher_->PublishActor(
             actor_id, *GenActorDataOnlyWithStates(*actor_table_data), nullptr));
+        RAY_CHECK_OK(gcs_table_storage_->ActorTaskSpecTable().Delete(actor_id, nullptr));
```

Comment:
There is an edge case where the actor is marked dead, but this delete doesn't finish. This will result in data leak. 
It's not a big deal, but we can check this case when loading data after GCS restarts.